<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Shengtong Han" />


<title>De novo mutations of ASD</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation-rate-modeling</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/mutation-rate-modeling/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>


<div id="header">



<h1 class="title toc-ignore">De novo mutations of ASD</h1>
<h4 class="author">Shengtong Han</h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<div id="samples" class="section level2">
<h2>Samples</h2>
<div id="case-samples" class="section level3">
<h3>Case Samples</h3>
<table>
<caption>Five WGS DNM case sample information</caption>
<thead>
<tr class="header">
<th>index</th>
<th>Samples</th>
<th># mutations</th>
<th># subjects</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Jiang_cases_DNM.bed</td>
<td>2,091</td>
<td>32</td>
</tr>
<tr class="even">
<td>2</td>
<td>Michaelson_cases_DNM.bed</td>
<td>581</td>
<td>10</td>
</tr>
<tr class="odd">
<td>3</td>
<td>Yuen_NM2015_cases_DNM.bed</td>
<td>9,381</td>
<td>162</td>
</tr>
<tr class="even">
<td>4</td>
<td>Wu_cases_DNM.bed</td>
<td>1,915</td>
<td>32</td>
</tr>
<tr class="odd">
<td>5</td>
<td>Kong_cases_DNM.bed</td>
<td>4,930</td>
<td>78</td>
</tr>
</tbody>
</table>
<ul>
<li><p>Jiang_cases_DNM.bed <a href="https://pubmed.ncbi.nlm.nih.gov/23849776/">Detection of clinically relevant genetic variants in autism spectrum disorder by whole-genome sequencing</a>, AJHG 2013.</p></li>
<li><p>Michaelson_cases_DNM.bed <a href="https://pubmed.ncbi.nlm.nih.gov/23260136/">Whole-genome sequencing in autism identifies hot spots for de novo germline mutation</a>, cell, 2012</p></li>
<li><p>Yuen_NM2015_cases_DNM.bed <a href="https://pubmed.ncbi.nlm.nih.gov/25621899/">Whole-genome sequencing of quartet families with autism spectrum disorder</a>, nature medicine, 2015</p></li>
<li><p>Wu_cases_DNM.bed [unpublished]</p></li>
<li><p>Kong_cases_DNM.bed <a href="https://www.nature.com/articles/nature11396">Rate of de novo mutations and the importance of father’s age to disease risk</a>, nature, 2012</p></li>
</ul>
</div>
<div id="wong_nc_2016_693_control_snv--control-sample" class="section level3">
<h3>Wong_NC_2016_693_control_SNV- Control sample</h3>
<table>
<caption>WGS DNM control sample information</caption>
<thead>
<tr class="header">
<th>index</th>
<th>Samples</th>
<th># mutations</th>
<th># subjects</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Wong_NC_2016_693_control_SNV.bed</td>
<td>27,092</td>
<td>693</td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;C</td>
<td>990 (3.65%)</td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;G</td>
<td>3,812 (14.07%)</td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>A-&gt;T</td>
<td>815 (3.01%)</td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;G</td>
<td>1,274 (4.70%)</td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;T</td>
<td>5,314 (19.61%)</td>
<td></td>
</tr>
<tr class="odd">
<td>7</td>
<td>C-&gt;A</td>
<td>1,298 (4.79%)</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>There are 585,017,944 C’s and 844,868,045 A’s in hg19.</p>
</div>
<div id="jonsson" class="section level3">
<h3>Jonsson</h3>
<p>De novo WGS of Jonsson data has 1548 controls.</p>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th># mutations</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C</td>
<td>3,582</td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G</td>
<td>13,477</td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T</td>
<td>3,331</td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G</td>
<td>4,822</td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T</td>
<td>21,220</td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A</td>
<td>3,862</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="simons-control" class="section level3">
<h3>Simons-control</h3>
<p>WGS Simons control data</p>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th># mutations</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C (+ T-&gt;G)</td>
<td>3,899+3,900=7,799</td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G (+ T-&gt;C)</td>
<td>15,083+14,919=30,002</td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T (+ T-&gt;A)</td>
<td>3,664+3,743=7,407</td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G (+ G-&gt;C)</td>
<td>5,393+5,308=10,701</td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T (+ G-&gt;A)</td>
<td>23,977+23,907=47,884</td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A (+ G-&gt;T)</td>
<td>5,404+5,652=11,056</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="yuen-data" class="section level3">
<h3>Yuen data</h3>
<p>Yuen data</p>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th># mutations</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C</td>
<td>4,239</td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G</td>
<td>15,893</td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T</td>
<td>4,086</td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G</td>
<td>5,433</td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T</td>
<td>24,418</td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A</td>
<td>5,942</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="sample-summary" class="section level3">
<h3>Sample summary</h3>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th>Wong-control</th>
<th>Jonsson</th>
<th>simons-control</th>
<th>Yuen</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C</td>
<td>990 (3.65%)</td>
<td>3,582</td>
<td>3,899</td>
<td>4,239</td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G</td>
<td>3,812 (14.07%)</td>
<td>13,477</td>
<td>15,083</td>
<td>15,893</td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T</td>
<td>815 (3.01%)</td>
<td>3,331</td>
<td>3,664</td>
<td>4,086</td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G</td>
<td>1,274 (4.70%)</td>
<td>4,822</td>
<td>5,393</td>
<td>5,433</td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T</td>
<td>5,314 (19.61%)</td>
<td>21,220</td>
<td>23,977</td>
<td>24,418</td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A</td>
<td>1,298 (4.79%)</td>
<td>3,862</td>
<td>5,404</td>
<td>5,942</td>
</tr>
</tbody>
</table>
<pre class="r"><code>Wong data: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Wong_NC_2016_693_control_SNV_with_allele_info.txt
Jonson: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Jonsson_Yuen2017/comb_SNPs_Jonsson
Yuen:  /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Jonsson_Yuen2017/comb_SNPs_Yuen
Simons: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Simons_WGS</code></pre>
<ul>
<li><p>Wong data: <a href="https://www.nature.com/articles/ncomms10486">New observations on maternal age effect on germline de novo mutation</a>, nature communications, 2016</p></li>
<li><p>Jonson, <a href="https://www.nature.com/articles/sdata2017115">Whole genome characterization of sequence diversity of 15,220 Icelanders</a>, Scientific Data, 2017</p></li>
<li><p>Simons: <a href="https://pubmed.ncbi.nlm.nih.gov/28965761/">Genomic Patterns of De Novo Mutation in Simplex Autism</a>, cell, 2017</p></li>
</ul>
<pre class="r"><code>Folder: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Simons_WGS/controls
Data for 100bp window: table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.ExpObsVar.bed.bed.add.100bp
Each row shows data of one 100bp window. Each column: chromosome, start, end, observed DNM count in the window, and fitted mutation rate in the window. Those counts are obtained by combing all mutation types together

chr1     0   100 0   0
chr1     100     200 0   0
chr1     200     300 0   0
chr1     300     400 0   0
chr1     400     500 0   0
chr1     500     600 0   0
chr1     600     700 0   0
chr1     700     800 0   0
chr1     800     900 0   0
chr1     900     1000 0      0</code></pre>
<div id="mutaton-count-by-mutationt-type" class="section level4">
<h4>mutaton count by mutationt type</h4>
</div>
</div>
</div>
<div id="data-processing-and-model-fitting" class="section level2">
<h2>Data processing and model fitting</h2>
<div id="data-discretization-and-categorization" class="section level3">
<h3>Data discretization and categorization</h3>
<ul>
<li>Continuous features are converted into binary discrete variables.<br />
</li>
<li>use R function Split to cluster windows with similar binary (original and converted) features.</li>
</ul>
</div>
<div id="model-fitting-for-each-mutation-type-separately" class="section level3">
<h3>Model fitting for each mutation type separately</h3>
<p>There are two ways to fit the model, GLM and optims. Both methods give the same esrimate either for whole genome or categorized windows as whown in simulations. Here only present the result by GLM with categorization,</p>
<ul>
<li><p>GLM (use R function GLM) with offset term</p>
<p><span class="math display">\[y\sim Pois( e^{\mu+X^T\beta})\]</span></p>
<ol style="list-style-type: decimal">
<li><p>With categorization, response <span class="math inline">\(y\)</span> is the <strong>total number of mutations</strong> from all windows within a category and covariate feature</p></li>
<li><p>offset term uses 7-mer ERV baseline mutation rate, which is sum of 7-mer mutation rate in a window for each mutation type, e.g. for A-&gt;C, sum over all 7-mer mutation rate with A-&gt;C in the center.</p></li>
<li><p>intercept is included and the same across all categories</p></li>
<li><p>confidence interval could be obtained as well</p></li>
</ol></li>
</ul>
</div>
</div>
<div id="erv-baseline-mutation-rate" class="section level2">
<h2>ERV baseline mutation rate</h2>
<table>
<caption>Average mutation rate of different mutation types in ERV paper.</caption>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th>mutations rate</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C</td>
<td><span class="math inline">\(1.6627\times 10^{-9}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G</td>
<td><span class="math inline">\(6.1805 \times 10^{-9}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T</td>
<td><span class="math inline">\(1.5633 \times 10^{-9}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G</td>
<td><span class="math inline">\(2.8075 \times 10^{-9}\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T</td>
<td><span class="math inline">\(1.2885 \times 10^{-8}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A</td>
<td><span class="math inline">\(3.3977 \times 10^{-9}\)</span></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="whole-genome-fit-14-features-simultaneously" class="section level2">
<h2>Whole genome-fit 14 features simultaneously</h2>
<p>The window length is set 100 bp. Use GLM to fit the model, and stated otherwise.</p>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-10-1.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-2.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-3.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-4.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-5.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-6.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-7.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-8.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-9.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-10.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-11.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-12.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-13.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-10-14.png" width="672" /></p>
<ul>
<li><p>Fit 14 covariate features simultaneously.</p></li>
<li><p>95% confidence interval by GLM is often big.</p></li>
</ul>
<div id="parameter-estimate-comparison" class="section level4">
<h4>Parameter estimate comparison</h4>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-13-1.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-2.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-3.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-4.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-5.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-6.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-7.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-8.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-9.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-10.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-11.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-12.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-13.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-13-14.png" width="672" /></p>
</div>
</div>
<div id="parameter-estimate-with-variable-selection" class="section level2">
<h2>Parameter estimate with variable selection</h2>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-16-1.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-2.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-3.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-4.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-5.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-6.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-7.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-8.png" width="672" /><img src="denovo-data_files/figure-html/unnamed-chunk-16-9.png" width="672" /></p>
<ul>
<li>compute the ratio of confidence interval over absolute value of estimated coefficient, rank them, and choose maximum of the top 25% quantile or 1. That is at least there are 25% covariate features with smallest ratios are selected as significant features.</li>
</ul>
</div>
<div id="methylation-feature-only" class="section level2">
<h2>methylation feature only</h2>
<ul>
<li>base level methylation are averaged in a window as window level methylation, then converted into binary variable with mean as threshold to fit the model</li>
<li>regress mutation count on methylation feature only</li>
<li>parameter estimated (<span class="math inline">\(\hat{\beta}\)</span>) + 95% CI (vertical line)</li>
<li>methylation samples are downloaded from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63818">GSE63818</a></li>
<li>methylation samples are strand specific. Plus strand data uses plus strand methylation sample and the same for minus strand.</li>
<li>methylation: GSE63818_Brain_5W_embryo1_methylation</li>
</ul>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p><img src="denovo-data_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<ul>
<li>For each mutation type, <span class="math inline">\(\widehat{\beta}\)</span> is nearly identical across difference methylation sample.</li>
</ul>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 18363)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.7        purrr_0.3.4       
##  [5] readr_2.1.1        tidyr_1.1.4        tibble_3.1.6       ggplot2_3.3.5     
##  [9] tidyverse_1.3.1    ape_5.6-1          gplots_3.1.1       RColorBrewer_1.1-2
## [13] kableExtra_1.3.4   knitr_1.37        
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.2         jsonlite_1.7.3     viridisLite_0.4.0  modelr_0.1.8      
##  [5] gtools_3.9.2       assertthat_0.2.1   highr_0.9          cellranger_1.1.0  
##  [9] yaml_2.2.2         pillar_1.6.5       backports_1.4.1    lattice_0.20-45   
## [13] glue_1.6.0         digest_0.6.29      rvest_1.0.2        colorspace_2.0-2  
## [17] htmltools_0.5.2    pkgconfig_2.0.3    broom_0.7.11       haven_2.4.3       
## [21] scales_1.1.1       webshot_0.5.2      svglite_2.0.0      tzdb_0.2.0        
## [25] farver_2.1.0       generics_0.1.1     ellipsis_0.3.2     withr_2.4.3       
## [29] cli_3.1.1          magrittr_2.0.1     crayon_1.4.2       readxl_1.3.1      
## [33] evaluate_0.14      fs_1.5.2           fansi_1.0.2        nlme_3.1-153      
## [37] xml2_1.3.3         tools_4.1.2        hms_1.1.1          lifecycle_1.0.1   
## [41] munsell_0.5.0      reprex_2.0.1       compiler_4.1.2     jquerylib_0.1.4   
## [45] caTools_1.18.2     systemfonts_1.0.3  rlang_0.4.12       grid_4.1.2        
## [49] rstudioapi_0.13    labeling_0.4.2     bitops_1.0-7       rmarkdown_2.11    
## [53] gtable_0.3.0       DBI_1.1.2          R6_2.5.1           lubridate_1.8.0   
## [57] fastmap_1.1.0      utf8_1.2.2         KernSmooth_2.23-20 stringi_1.7.6     
## [61] parallel_4.1.2     Rcpp_1.0.8         vctrs_0.3.8        dbplyr_2.1.1      
## [65] tidyselect_1.1.1   xfun_0.29</code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
