<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Shengtong Han" />


<title>Fitting Diagnostic</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation-rate-modeling</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/mutation-rate-modeling/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>


<div id="header">



<h1 class="title toc-ignore">Fitting Diagnostic</h1>
<h4 class="author">Shengtong Han</h4>
<h4 class="date">YYYY-MM-DD</h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<div id="simons-control" class="section level2">
<h2>Simons-control</h2>
<div id="bp-windows" class="section level3">
<h3>100 bp windows</h3>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th># total windows</th>
<th># window <span class="math inline">\(p.val&lt;0.05\)</span></th>
<th># window <span class="math inline">\(p.val=0\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C</td>
<td>26,839,568</td>
<td>3,899</td>
<td>3,870</td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G</td>
<td>15,067</td>
<td>15,067</td>
<td>10,806</td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T</td>
<td>26,839,598</td>
<td>3,656</td>
<td>3,649</td>
</tr>
</tbody>
</table>
<pre class="r"><code>Data: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Simons_WGS/controls/table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.ExpObsVar.bed.bed.add.100bp
Columns: chr, start, end, observed rate, fitted rate.</code></pre>
</div>
<div id="k-windows" class="section level3">
<h3>5k windows</h3>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<ul>
<li><p>aggregate summary statistics, expected rate and observed count in 100 bp windows</p></li>
<li><p>there are 2 windows with expected rate &gt;100</p></li>
</ul>
<pre><code>&gt; ObsandExp[which(as.numeric(as.character(ObsandExp$V5))&gt;100),]
# A tibble: 2 x 5
  V1           V2        V3 V4    V5         
  &lt;chr&gt;     &lt;int&gt;     &lt;int&gt; &lt;chr&gt; &lt;chr&gt;      
1 chr8  139250000 139255000 0     166.3652155
2 chr8  139255000 139260000 0     101.249892</code></pre>
<p>See below the high recombination rate of two 100 bp windows.</p>
<pre><code>chr8     139249900       139250000       0       5.5363439e-08   0       0       1       0.44    .       108.804         0       0         1       0       0       0       0       0
chr8     139250000       139250100       0       3.779853e-08    0       0       1       0.51    -1.174439       108.804         0         0       1       0       0       0       0       0</code></pre>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="k-windows-1" class="section level3">
<h3>10k windows</h3>
<p>add 100bp windows: Observed and estimated mutations for 100k windows are obtained by adding up that in 100bp windows.</p>
<div id="residual-distribution-and-qq-plot-of-p-values" class="section level4">
<h4>Residual distribution and QQ plot of p values</h4>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-7-9.png" width="672" /></p>
</div>
<div id="windows-with-high-expected-mutation-rate" class="section level4">
<h4>Windows with high expected mutation rate</h4>
<table>
<caption>
windows with standardized residual less than -5
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:left;">
observed
</th>
<th style="text-align:left;">
expected
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A_to_C_1
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139240000
</td>
<td style="text-align:right;">
139250000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
189.8829699
</td>
</tr>
<tr>
<td style="text-align:left;">
A_to_C_2
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139250000
</td>
<td style="text-align:right;">
139260000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
650.7239649
</td>
</tr>
<tr>
<td style="text-align:left;">
C_to_A.CpG_1
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139240000
</td>
<td style="text-align:right;">
139250000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
45.33371804
</td>
</tr>
<tr>
<td style="text-align:left;">
C_to_A.CpG_2
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139250000
</td>
<td style="text-align:right;">
139260000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
137.5672679
</td>
</tr>
<tr>
<td style="text-align:left;">
C_to_T.CpG_1
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139250000
</td>
<td style="text-align:right;">
139260000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
42.86830912
</td>
</tr>
</tbody>
</table>
<p>The reason for high expected mutation rate is that there are about 100 windows of 100 bp that have recombination rates as high as 108.</p>
</div>
<div id="hotspot-detection-and-neighborhood-correlation" class="section level4">
<h4>hotspot detection and neighborhood correlation</h4>
<p>One way to define hotsopt is look at standardized residuals that are greater than a threshold, say 4 and study the distribution of selected ‘hotspot’ windows.</p>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-10-9.png" width="672" /></p>
<ul>
<li>No.consecutive windows: 0 means ‘isolated’ windows , 1 means two-consecutive windows, and etc.<br />
</li>
<li>with cutoff of 4, all “hotsopt” are isolated and consecutive hotspots are clearly observed until cutoff reduces to 2.</li>
</ul>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-11-9.png" width="672" /></p>
<ul>
<li>for every hotspot window with a residual cutoff, look at the distance to its nearest neighbor and plot the histgram of distance and density</li>
<li>the mean of the distance is shown in each graph and the blue vertical line is reference line with distance of 1.</li>
<li>red line are density of sample distribution and green line is pdf of null distribution of random windows</li>
<li>p.val is calculated by ks.test for testing if two distributions are identical or not</li>
</ul>
</div>
<div id="scatter-plot" class="section level4">
<h4>scatter plot</h4>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<ul>
<li>Every point is one chromosome where expected mutation rate and observed mutation count in each 10k window are added up.</li>
</ul>
</div>
</div>
<div id="k-windows-2" class="section level3">
<h3>100k windows</h3>
<p>add 100bp windows: Observed and estimated mutations for 100k windows are obtained by adding up that in 100bp windows.</p>
<div id="residual-distribution-and-qq-plot-of-p-values-1" class="section level4">
<h4>Residual distribution and QQ plot of p values</h4>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-13-9.png" width="672" /></p>
</div>
<div id="windows-with-high-expected-mutation-rate-1" class="section level4">
<h4>Windows with high expected mutation rate</h4>
<table>
<caption>
windows with standardized residual less than -5
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:left;">
observed
</th>
<th style="text-align:left;">
expected
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A_to_C
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139200000
</td>
<td style="text-align:right;">
139300000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
840.7832083
</td>
</tr>
<tr>
<td style="text-align:left;">
C_to_A.CpG
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139200000
</td>
<td style="text-align:right;">
139300000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
182.9181968
</td>
</tr>
<tr>
<td style="text-align:left;">
C_to_T.CpG
</td>
<td style="text-align:left;">
chr8
</td>
<td style="text-align:right;">
139200000
</td>
<td style="text-align:right;">
139300000
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
61.42456014
</td>
</tr>
</tbody>
</table>
</div>
<div id="hotspot-detection-and-neighborhood-correlation-1" class="section level4">
<h4>hotspot detection and neighborhood correlation</h4>
<p>One way to define hotsopt is look at standardized residuals that are greater than a threshold, say 4 and study the distribution of selected ‘hotspot’ windows.</p>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-17-9.png" width="672" /></p>
<ul>
<li><p>No.consecutive windows: 0 means ‘isolated’ windows , 1 means two-consecutive windows, and etc.<br />
</p></li>
<li><p>with cutoff of 4, all “hotsopt” are isolated and consecutive hotspots are clearly observed until cutoff reduces to 2.</p></li>
<li><p>points over the bars are -log(p values) of enrichment with background, -log(p values) are forced to be 5 when they are greater than 5.</p></li>
<li><p>In every table, each column is the residual cutoff and each row is the number of consecutive windows.<br />
</p></li>
<li><p>Numbers in the table are p values: In the whole genome, randomly pick the same amount of random windows as hotspot windows, do a enrichment test between hotspot windows and random windows under each scenario by one-side poisson test.</p></li>
</ul>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-1.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-2.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-3.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-4.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-5.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-6.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-7.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-8.png" width="672" /><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-20-9.png" width="672" /></p>
<ul>
<li>for every hotspot window with a residual cutoff, look at the distance to its nearest neighbor and plot the histgram of distance and density</li>
<li>the mean of the distance is shown in each graph and the blue vertical line is reference line with distance of 1.</li>
<li>red line are density of sample distribution and green line is pdf of null distribution of random windows</li>
<li>p.val is calculated by ks.test for testing if two distributions are identical or not</li>
</ul>
</div>
<div id="scatter-plot-1" class="section level4">
<h4>scatter plot</h4>
<p><img src="FittingDiagnostic_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<ul>
<li>Every point is one chromosome where expected mutation rate and observed mutation count in each 100k window are added up.</li>
</ul>
</div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
