<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Shengtong Han" />


<title>Modeling</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation-rate-modeling</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/mutation-rate-modeling/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Modeling</h1>
<h4 class="author"><em>Shengtong Han</em></h4>
<h4 class="date"><em>YYYY-MM-DD</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-04-10</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 3ae630c</p>
<!-- Add your analysis here -->
<div id="session-information" class="section level2">
<h2>Session information</h2>
</div>
<div id="base-level" class="section level2">
<h2>Base level</h2>
<p>At every genomic locus, observed number of mutations, <span class="math inline">\(y_i\)</span> is modeled as Poisson distribution as</p>
<p><span class="math inline">\(y_i \sim Pois(\mu_i)\)</span></p>
<p>There are several covariates affecting the mutation rate, such as tri-nucleotide context, parents’ age, etc. To relate the mutation rate to covariates <span class="math inline">\(X_i\)</span>, Poisson regression could be one way to do that</p>
<p><span class="math inline">\(log(E(y_i|X_i))=log(\mu_i)=X_i^T\beta\)</span></p>
<p>Thus</p>
<p><span class="math inline">\(\mu_i=e^{X_i^T\beta}\)</span>.</p>
<p>The likelihood function is</p>
<p><span class="math display">\[
\begin{aligned} 
L(\beta|Y)&amp;=\prod_i\frac{[e^{X_i^T\beta}]^{Y_i}e^{-e^{X_i^T\beta}}}{Y_i!}\\
&amp;=e^{\sum_i[X_i^T\beta Y_i-e^{X_i^T\beta}]} \times constant
\end{aligned}
\]</span></p>
<p>The log likelihood is</p>
<p><span class="math display">\[
\begin{aligned}
l(\beta)&amp;=\sum_i[X_i^T\beta Y_i-e^{X_i^T\beta}]+constant\\
&amp;=\sum_i(X_i^T\beta)Y_i-\sum_ie^{X_i^T\beta}+constant
\end{aligned}
\]</span></p>
</div>
<div id="region-level" class="section level2">
<h2>Region level</h2>
<p>It’s computationally hard to infer the mutation rate at single base since there are 3B bases in human genome. So consider the window level. Within a window, collapse all bases together, denote <span class="math inline">\(Y\)</span> by the total number of mutations, assume <span class="math inline">\(X_i\)</span> are the same across all bases. Within a window, assuming <span class="math inline">\(Y_i\sim Pois(\mu_i)\)</span> are indepedent poisson random variables, the sum <span class="math inline">\(Y=\sum_iY_i \sim Pois (\sum_i\mu_i)\)</span></p>
<p>Considering baseline mutation rate <span class="math inline">\(\mu_{i0}\)</span> at every base <span class="math inline">\(i\)</span>, then <span class="math inline">\(Y \sim Pois(\sum_i \mu_{i0} \mu_i)\)</span>, where <span class="math inline">\(\sum_i \mu_{i0}\mu_i=\sum_i\mu_{i0}e^{X_i^T\beta}=e^{X^T\beta}\sum_i\mu_{i0}\)</span>, assuming <span class="math inline">\(x_i^T\beta\)</span> are the same across all bases in the window and <span class="math inline">\(\sum_i \mu_{i0}\)</span> is the total baseline mutation for the window. Then <span class="math inline">\(Y_j \sim Pois(\mu_{j0}e^{X_j^T\beta})\)</span></p>
<p>The likelihood funciton is</p>
<p><span class="math display">\[
\begin{aligned} 
L(\beta|Y)=\prod_j\frac{[\mu_{j0}e^{X_j^T\beta}]^{Y_j}e^{-\mu_{j0}e^{X_j^T\beta}}}{Y_j!}
\end{aligned}
\]</span></p>
<p>and</p>
<ul>
<li><span class="math inline">\(Y_j\)</span>; total number of mutations in window <span class="math inline">\(j\)</span></li>
<li><span class="math inline">\(X_j\)</span>: a vector of covariates for the window, assuming the same acorss all bases in the window</li>
</ul>
<p>And the log likelihood function is</p>
<p><span class="math display">\[
\begin{aligned}
l(\beta)=\sum_j[Y_jlog(\mu_{j0}e^{X_j^T\beta})]-\sum_j[\mu_{j0}e^{X_j^T\beta}]+constant
\end{aligned}
\]</span></p>
</div>
<div id="categorization" class="section level2">
<h2>Categorization</h2>
<p>Human genome has 3B bases, even for 100 bp windows, the computation is still expensive. To speed up, one way is to put all windows with similar features into one category and treat the whole category as a single unit. Suppose we have <span class="math inline">\(P\)</span> binary features (continuous features could be discretized into discrete values). Totally it is going to be <span class="math inline">\(2^P\)</span> feature categories. Usually <span class="math inline">\(2^P &lt;&lt; \frac{3B}{L}\)</span>, <span class="math inline">\(L\)</span> is window length.</p>
<p>Let <span class="math inline">\(c\)</span> index category. <span class="math inline">\(Y_c\)</span> is the total number of mutations, <span class="math inline">\(\mu_{c0}\)</span> is the baseline mutation rate, <span class="math inline">\(X_c\)</span> is the feature covariates.</p>
<p>The likelihood function is</p>
<p><span class="math display">\[
\begin{aligned}
 logP(Y|\beta) &amp; = \sum_j[Y_jlog(\mu_{j0}e^{X_j^T\beta})]-\sum_j[\mu_{j0}e^{X_j^T\beta}]+constant(following ~LH~of~region~level)\\
 &amp;=\sum_c\{Y_c(X_c^T\beta)-\mu_{c0}e^{X_c^T\beta} \}+constant
\end{aligned}
\]</span></p>
<ul>
<li><span class="math inline">\(\mu_{c0}=\sum_j\mu_{j0}\)</span> in category <span class="math inline">\(c\)</span>.</li>
<li>The last equation is because for category <span class="math inline">\(c\)</span>, all <span class="math inline">\(X_j=X_c\)</span> are the same, so <span class="math inline">\(\sum_jY_j(X_j^T\beta)=X_c^T\beta \sum_jY_j=Y_c(X_c^T\beta)\)</span>.</li>
</ul>
<p>Thus <span class="math inline">\(Y_c\sim Pois(\mu_{c0}e^{X_c^T\beta})\)</span>. Another way to understand it is that since <span class="math inline">\(Y_j \sim Pois(\mu_{j0}e^{X_j^T\beta})\)</span>, in category <span class="math inline">\(c\)</span>, the sum of <span class="math inline">\(\sum_j Y_j \sim Pois(\sum_j \mu_{j0}e^{X_j^T\beta})\)</span>, where <span class="math inline">\(\sum_j \mu_{j0}e^{X_j^T\beta}=e^{X_c\beta} \sum_j \mu_{j0}=\mu_{c0}e^{X_c\beta}\)</span>, assuming <span class="math inline">\(Y_j\)</span> are independent.</p>
<div id="likelihood-and-derivative" class="section level3">
<h3>Likelihood and derivative</h3>
<p>Let <span class="math inline">\(f(\beta)=\sum_c\{Y_c(X_c^T\beta)-\mu_{c0}e^{X_c^T\beta} \}\)</span> be the objective function to be optimized over <span class="math inline">\(\beta\)</span>. <span class="math inline">\(\frac{\partial f(\beta)}{\partial \beta_i}=\sum_c (Y_c(X_{ic})-\mu_{c0}X_{ic}e^{X_c\beta})\)</span>, <span class="math inline">\(i=1, \cdots, P\)</span>.</p>
</div>
</div>
<div id="simulations" class="section level2">
<h2>Simulations</h2>
<div id="constant-baseline-mutation-rate-mu_0" class="section level3">
<h3>Constant baseline mutation rate <span class="math inline">\(\mu_0\)</span></h3>
<div id="use-true-baseline-mutation-rate-as-offset" class="section level4">
<h4>Use true baseline mutation rate as offset</h4>
<pre class="r"><code>#rm(list=ls())
set.seed(1000)
n &lt;- 10000
baseline.rate=0.01
#regression coefficients
 beta0 &lt;- 0.1
 beta1 &lt;- log(2)
#generate covariate values
 x &lt;- rbinom(n, 1, prob=0.5)
#compute mu&#39;s
mu &lt;- exp(beta0 + beta1 * x)*baseline.rate
#generate Y-values
 y &lt;- rpois(n=n, lambda=mu)
   #data set
 data &lt;- data.frame(y=y, x=x)
 y0=sum(data$y[data$x==0]); y1=sum(data$y[data$x==1])
 cate.data=data.frame(y=c(y0,y1), x=c(0,1))
 fit=glm(data$y~data$x+offset(log(rep(baseline.rate, n))), family=poisson())
 cate.fit=glm(cate.data$y~cate.data$x+offset(log(c(baseline.rate*sum(x==cate.data$x[1]), baseline.rate*sum(x==cate.data$x[2])))), family=poisson())
fit.conf=confint(fit)</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<pre class="r"><code>cate.fit.conf=confint(cate.fit)</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<p><img src="figure/modeling.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>sample size, i.e. number of windows <span class="math inline">\(N=10^4\)</span></li>
<li>baseline rate is constant across all windows <span class="math inline">\(\mu_0=0.01\)</span>.</li>
<li><span class="math inline">\(\beta_0=0.1, \beta_1=log(2)\)</span>.</li>
</ul>
<pre><code>Waiting for profiling to be done...
Waiting for profiling to be done...</code></pre>
<p><img src="figure/modeling.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>sample size, i.e. number of windows <span class="math inline">\(N=10^6\)</span></li>
<li>baseline rate is constant across all windows <span class="math inline">\(\mu_0=0.01\)</span>.</li>
<li><span class="math inline">\(\beta_0=0.1, \beta_1=log(2)\)</span>.</li>
</ul>
<p><img src="figure/modeling.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>when use true baseline mutation rate, GLM with categorization and without give mostly identical results and both results are approaching to their true values when <span class="math inline">\(n \geq 10^6\)</span>.</li>
</ul>
</div>
<div id="use-estimated-baseline-mutation-rate" class="section level4">
<h4>Use estimated baseline mutation rate</h4>
<p>In most cases, we never know the true baseline mutation rate, so we may have to use their estimates</p>
<pre class="r"><code>fit=glm(data$y~data$x+offset(log(rep(sum(data$y)/n, n))), family=poisson())
cate.fit=glm(cate.data$y~cate.data$x+offset(log(c(sum(cate.data$y)/n*sum(x==cate.data$x[1]), sum(cate.data$y)/n*sum(x==cate.data$x[2])))), family=poisson())</code></pre>
<p><img src="figure/modeling.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>with estimated baseline mutation rate as offset, intercept esmtiate is not correct, but <span class="math inline">\(\widehat{\beta_1}\)</span> is good.</li>
</ul>
</div>
</div>
<div id="variable-baseline-mutatiom-rate-mu_i0" class="section level3">
<h3>Variable baseline mutatiom rate <span class="math inline">\(\mu_{i0}\)</span></h3>
<div id="use-true-baseline-mutation-rate" class="section level4">
<h4>use true baseline mutation rate</h4>
<pre><code>Waiting for profiling to be done...
Waiting for profiling to be done...</code></pre>
<p><img src="figure/modeling.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><span class="math inline">\(\mu_{i0}\)</span> is generated by <span class="math inline">\(\mu_{i0} \sim rbeta(n, 1, 100)\)</span></li>
</ul>
<p><img src="figure/modeling.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="use-estimated-baseline-mutation-rate-1" class="section level4">
<h4>use estimated baseline mutation rate</h4>
<pre class="r"><code>fit=glm(data$y~data$x+offset(log(rep(sum(data$y)/n, n))), family=poisson())
cate.fit=glm(cate.data$y~cate.data$x+offset(log(c(sum(cate.data$y)/n*sum(x==cate.data$x[1]), sum(cate.data$y)/n*sum(x==cate.data$x[2])))), family=poisson())</code></pre>
<p><img src="figure/modeling.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>with estimated baseline mutation rate as offset, intercept esmtiate is not correct, but <span class="math inline">\(\widehat{\beta_1}\)</span> is good.</li>
</ul>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.4.3  backports_1.0.5 magrittr_1.5    rprojroot_1.3-1
 [5] tools_3.4.3     htmltools_0.3.5 yaml_2.1.16     Rcpp_0.12.10   
 [9] stringi_1.1.5   rmarkdown_1.8   knitr_1.17      git2r_0.21.0   
[13] stringr_1.2.0   digest_0.6.12   evaluate_0.10  </code></pre>
</div>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
