<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Simons control WGS</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mutation-rate-modeling</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://han16.github.io/mutation-rate-modeling/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Simons control WGS</h1>

</div>


<div id="mutation-count-partition-by-mutation-types" class="section level2">
<h2>Mutation count partition by mutation types</h2>
<p>WGS Simons control data: 516 idiopathic autism families (2,064 individuals)</p>
<table>
<thead>
<tr class="header">
<th>index</th>
<th>mutation type</th>
<th># mutations</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>A-&gt;C (+ T-&gt;G)</td>
<td>3,899+3,900=7,799</td>
<td></td>
</tr>
<tr class="even">
<td>2</td>
<td>A-&gt;G (+ T-&gt;C)</td>
<td>15,083+14,919=30,002</td>
<td></td>
</tr>
<tr class="odd">
<td>3</td>
<td>A-&gt;T (+ T-&gt;A)</td>
<td>3,664+3,743=7,407</td>
<td></td>
</tr>
<tr class="even">
<td>4</td>
<td>C-&gt;G (+ G-&gt;C)</td>
<td>5,393+5,308=10,701</td>
<td></td>
</tr>
<tr class="odd">
<td>5</td>
<td>C-&gt;T (+ G-&gt;A)</td>
<td>23,977+23,907=47,884</td>
<td></td>
</tr>
<tr class="even">
<td>6</td>
<td>C-&gt;A (+ G-&gt;T)</td>
<td>5,404+5,652=11,056</td>
<td></td>
</tr>
</tbody>
</table>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="model-evalulation" class="section level2">
<h2>Model evalulation</h2>
<ul>
<li><p>Model 0 (M0): 3-mer only for 9 mutation types. This is implemented by enforcing all parameters to be zero in M1.</p></li>
<li><p>Model 1 (M1): 3-mer+features for pooled 9 mutation types</p></li>
<li><p>Model 2 (M2): 3-mer+features for 9 mutation types</p></li>
</ul>
<div id="predictive-density-measure" class="section level3">
<h3>Predictive density measure</h3>
<ul>
<li>Whole human genome (hg19) is evenly split into 100bp windows</li>
<li>use even (odd) windows as training data to estimate parameters <span class="math inline">\(\beta\)</span></li>
<li>use another half odd (even) windows as validation data to make prediction</li>
<li>windows with predicted density as <code>NaN</code> due to <span class="math inline">\(\widehat{\mu}_j=0\)</span> are removed</li>
<li>M0 uses scale parameters to calculate predicted rate</li>
<li>Overall these two models are comparable to each other, none of which outperforms another across all mutation types. For some mutation types, M0 works better and for others, M2 is better.</li>
</ul>
<div id="m0-vs-m2-by-mutation-types" class="section level4">
<h4>M0 vs M2 by mutation types</h4>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-4-1.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
</div>
<div id="predicted-probability-of-three-models" class="section level4">
<h4>predicted probability of three models</h4>
<ul>
<li><p>In M0, M2 for each one of 9 mutation types, use even(odd) window as train set to estimate parameters, and use remaining odd(even) window to calculate predicted probability, and sum up across all mutation types. In M1, use odd(even) window as train set, and remaining even(odd) window to calculate predicted probability and add them up.</p></li>
<li><p>the higher predicted probability, the better, M1&gt;M0&gt;M2</p></li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
</div>
<div id="expected-rate-vs-observed-rate" class="section level3">
<h3>Expected rate vs observed rate</h3>
<div id="m0-vs-m2-by-pooling-all-mutation-types" class="section level4">
<h4>M0 vs M2 by pooling all mutation types</h4>
<ul>
<li>each dot is one 100kb window by summing up 100bp windows</li>
<li>For each one of 9 mutation types, use odd(even) windows to estimate parameters, calculate expected rate in the same odd (even) window and sum up over all mutation types.</li>
<li>exp in M0 is expected rate adjusted by scaling parameter <span class="math inline">\(L=\frac{\sum y_i}{\sum \mu_i}\)</span></li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-6-1.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-6-2.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-6-3.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-6-4.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-6-5.png" width="672" /></p>
</div>
<div id="outlier-diagnostic" class="section level4">
<h4>Outlier diagnostic</h4>
<p>Note: In a few windows, the calculated expected rate could be unexpected high due to big recombination rate in several windows, such as 108.8.</p>
<p>Look at mutation type <code>mutation.A_to_C</code> at chromosome 8 as an example.</p>
<ul>
<li>recombination rate in those windows and across all windows</li>
</ul>
<pre><code> awk &#39;$11&gt;100&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.chr8.ERVbaseline.1DHS.2CpGisland.3lamin.4gc.5reptime.6RecombRate.7Exon.8H3K27ac.9H3K27me3.10H3K36me3.11H3K4me1.12H3K4me3.13H3K9ac.14H3K9me3.even.line
chr8     139247700       139247800       0       5.8508268e-08   0       0       1       0.41    .       108.804         0 0       1       0       0       0       0       1
chr8     139247900       139248000       0       3.7592286e-08   0       0       1       0.59    .       108.804         0 0       1       0       0       0       0       1
chr8     139248100       139248200       0       4.8498557e-08   0       0       1       0.44    .       108.804         0 0       1       0       0       0       0       1
</code></pre>
<p>there are 50 such windows in total and 50 in complementary strand <code>mutation.T_to_G</code>.</p>
<p>Those 50 windows have recombination rate (RR) of <code>108.804</code>, more than 100 times high than average RR across all windows of 0.84.</p>
<pre><code>awk &#39;{sum+=$11} END {print sum}&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.chr8.ERVbaseline.1DHS.2CpGisland.3lamin.4gc.5reptime.6RecombRate.7Exon.8H3K27ac.9H3K27me3.10H3K36me3.11H3K4me1.12H3K4me3.13H3K9ac.14H3K9me3.even.line
615910

wc -l  table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.chr8.ERVbaseline.1DHS.2CpGisland.3lamin.4gc.5reptime.6RecombRate.7Exon.8H3K27ac.9H3K27me3.10H3K36me3.11H3K4me1.12H3K4me3.13H3K9ac.14H3K9me3.even.line
731820 table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.chr8.ERVbaseline.1DHS.2CpGisland.3lamin.4gc.5reptime.6RecombRate.7Exon.8H3K27ac.9H3K27me3.10H3K36me3.11H3K4me1.12H3K4me3.13H3K9ac.14H3K9me3.even.line

&gt; 615910/731820
[1] 0.8416141
</code></pre>
<ul>
<li>predicted rate in those windows and across all windows</li>
</ul>
<p>all those 50 windows have predicted rate more than 0.5</p>
<pre><code>awk &#39;$5&gt;0.5&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.bed
chr8 139247700 139247800 0 0.885879142920415 0.000179683336383749 3927.62074039213 -0.941211529317621 0.826701765213676
chr8 139247900 139248000 0 0.514188561224198 0.000103058730615795 2306.96721407905 -0.717069425665464 0.763334370034514
chr8 139248100 139248200 0 0.814619465876907 0.000164902137512123 3618.84938902892 -0.902562721298031 0.816620990462835
chr8 139248300 139248400 0 0.845005809651171 0.000171279641853248 3748.88094484896 -0.919241975570726 0.821015489926893
chr8 139248500 139248600 0 0.851571372578056 0.000173296884058543 3763.06476062817 -0.922806248666564 0.821945908060196


awk &#39;$5&gt;0.5&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.bed | wc -l
50</code></pre>
<p>however M0 gives predicted rate</p>
<pre><code>grep &#39;chr8&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.with.zero.coefficient.and.intercept.bed | awk &#39;$2&gt;=139247700 &amp;&amp;  $2&lt;=139257500&#39;
chr8 139247700 139247800 0 1.00172949e-07 0.000179683336383749 3927.62074039213 -0.000316501104263476 0.500126265670176
chr8 139247900 139248000 0 5.868529e-08 0.000103058730615795 2306.96721407905 -0.000242250469555788 0.500096643953808
chr8 139248100 139248200 0 9.225769e-08 0.000164902137512123 3618.84938902892 -0.000303739510106934 0.500121174530947
chr8 139248300 139248400 0 9.5600379e-08 0.000171279641853248 3748.88094484896 -0.000309193109560999 0.500123350202247
chr8 139248500 139248600 0 9.6045573e-08 0.000173296884058543 3763.06476062817 -0.000309912202083106 0.500123637078644

grep &#39;chr8&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.with.zero.coefficient.and.intercept.bed | awk &#39;$2&gt;=139247700 &amp;&amp;  $2&lt;=139257500&#39; | wc -l
50
</code></pre>
<p>and adjusted rate is <code>3106*10^{-7}=10^{-4}</code>. Those 50 windows are 1000 times higher than that in M0 for predicted rate.</p>
<pre><code>awk &#39;{sum+=$4} END {print sum}&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.with.zero.coefficient.and.intercept.bed
3922

awk &#39;{sum+=$5} END {print sum}&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.with.zero.coefficient.and.intercept.bed
1.2627

&gt; 3922/1.2627
[1] 3106.043
</code></pre>
<ul>
<li>verify calculation of predicted rate</li>
</ul>
<pre><code>&gt; load(&quot;table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.100bpwindow.ev&gt; .line.train.model.RData&quot;)
&gt;
&gt; ls()
[1] &quot;result&quot;
&gt; result
$para.est
                                          (Intercept)
                                          7.946130900
      second.cate.matrix[, 3:(2 + feature_number)]DHS
                                          0.111412210
second.cate.matrix[, 3:(2 + feature_number)]CpGisland
                                         -0.061066892
    second.cate.matrix[, 3:(2 + feature_number)]Lamin
                                          0.049454107
       second.cate.matrix[, 3:(2 + feature_number)]GC
                                         -0.051563344
       second.cate.matrix[, 3:(2 + feature_number)]RT
                                          0.099047740
       second.cate.matrix[, 3:(2 + feature_number)]RR
                                          0.072775740
     second.cate.matrix[, 3:(2 + feature_number)]Exon
                                          0.033057750
  second.cate.matrix[, 3:(2 + feature_number)]H3K27ac
                                         -0.006163996
 second.cate.matrix[, 3:(2 + feature_number)]H3K27me3
                                          0.005764340
 second.cate.matrix[, 3:(2 + feature_number)]H3K36me3
                                          0.044806480
  second.cate.matrix[, 3:(2 + feature_number)]H3K4me1
                                          0.041314267
  second.cate.matrix[, 3:(2 + feature_number)]H3K4me3
                                          0.094146882
   second.cate.matrix[, 3:(2 + feature_number)]H3K9ac
                                          0.006397160
  second.cate.matrix[, 3:(2 + feature_number)]H3K9me3
                                          0.096692925
</code></pre>
<pre class="r"><code>coff.est=c(7.946130900, 0.111412210, -0.061066892,  0.049454107, -0.051563344, 0.099047740, 0.072775740, 0.033057750, -0.006163996, 0.005764340,0.044806480, 0.041314267, 0.094146882, 0.006397160, 0.096692925)
cova.fea.a2c=c(0,5.8508268e-08,0,0,1,0.41,0,108.804,0,0,1,0,0,0,0,1) # covariate feature for window chr8     139247700       139247800 for mutation A-&gt;C
cova.fea.t2g=c(0,4.1664681e-08,0,0,1,0.41,0,108.804,0,0,1,0,0,0,0,1)
#covariate feature for window chr8     139247700       139247800 for mutation T-&gt;G
feature_number=14
exp.rate=exp(cova.fea.a2c[3:(3+feature_number-1)]%*%coff.est[2:15]+coff.est[1])*(cova.fea.a2c[2]+cova.fea.t2g[2]) # this is expected rate
# covariate features are identical in both strands, but baseline rate should be added up in cauculating prediction rate  

cat(&quot;expected rate is&quot;, exp.rate, &quot;\n&quot;) # which is exactly the one in file </code></pre>
<pre><code>## expected rate is 0.8858791</code></pre>
<pre class="r"><code># awk &#39;$5&gt;0.5&#39; table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.A_to_C.and.mutation.T_to_G.ExpObsVar.even.window.as.train.bed
#chr8 139247700 139247800 0 0.885879142920415 0.000179683336383749 3927.62074039213 -0.941211529317621 0.826701765213676

# what happens if replace 108.8 with average rate of 0.8 

cova.fea.a2c=c(0,5.8508268e-08,0,0,1,0.41,0,0.8,0,0,1,0,0,0,0,1) 
cova.fea.t2g=c(0,4.1664681e-08,0,0,1,0.41,0,0.8,0,0,1,0,0,0,0,1)
exp.rate=exp(cova.fea.a2c[3:(3+feature_number-1)]%*%coff.est[2:15]+coff.est[1])*(cova.fea.a2c[2]+cova.fea.t2g[2]) # this is expected rate
cat(&quot;after replace 108 with 0.8, expected rate is&quot;, exp.rate, &quot;\n&quot;)</code></pre>
<pre><code>## after replace 108 with 0.8, expected rate is 0.0003418133</code></pre>
</div>
<div id="chromosome-wide-scatter-plot-by-combining-all-mutation-types-in-m0-m2" class="section level4">
<h4>Chromosome wide scatter plot by combining all mutation types in M0, M2</h4>
<ul>
<li>each dot is one chromosome by summing up across 100bp windows</li>
<li>M0, M2 use odd(even) windows to estimate parameters, then calculate expected rate in the same odd(even) windows and sum up for each one of 9 mutation types. M1 use odd(even) windows for parameter estimation and calculate expected rate in the same window and add up.</li>
<li>M0, M2 sum over all mutation types, M1 runs directly on mutation type combined sample<br />
</li>
<li>observed rate on y axis and expected on x axis</li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<ul>
<li><span class="math inline">\(MSE=\sum\frac{(obs-exp)^2}{22}\)</span></li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="parameter-estimate-in-m1-m2" class="section level2">
<h2>Parameter estimate in M1 &amp; M2</h2>
<div id="m1-vs-m2" class="section level3">
<h3>M1 vs M2</h3>
<ul>
<li><p>three continuous variables are firstly standardized, then converted into binary variable before collapsing in both M1 and M2 model</p></li>
<li><p>estimate in M1 is more robust than M2 in the sense that both odd and even windows produce similar estimates, but they could vary a lot across specific mutation types in M2</p></li>
<li><p><code>combine</code> highlighted by dashed horizontal lines denotes M1 model for pooled mutation types vs other 9 mutation types in M2 model on x axis</p></li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-1.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-2.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-3.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-4.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-5.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-6.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-7.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-8.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-9.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-10.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-11.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-12.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-13.png" width="672" /><img src="simons-control-wgs_files/figure-html/unnamed-chunk-15-14.png" width="672" /></p>
</div>
<div id="parameter-estimated-in-m1" class="section level3">
<h3>Parameter estimated in M1</h3>
<ul>
<li>three continuous variables are firstly standardized, then converted into binary variable before collapsing</li>
<li>vertical lines are 95% confidence interval</li>
</ul>
<p><img src="simons-control-wgs_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
</div>
<div id="referecnce" class="section level2">
<h2>Referecnce</h2>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/28965761/">Genomic Patterns of De Novo Mutation in Simplex Autism</a>, cell, 2017</p>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
