---
title: "De novo mutations of ASD"
author: "Shengtong  Han"
date: YYYY-MM-DD
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
set.seed(123)
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r, echo=F}
loadRData <- function(fileName){
#loads an RData file, and returns it
    load(fileName)
    get(ls()[ls() != "fileName"])
}
```


## Samples



### Case Samples 



index    | Samples | \# mutations| \# subjects
----|-------|-------|---
1 | Jiang_cases_DNM.bed | 2,091| 32
2 | Michaelson_cases_DNM.bed |581 | 10
3 | Yuen_NM2015_cases_DNM.bed | 9,381| 162
4 | Wu_cases_DNM.bed | 1,915 | 32
5 | Kong_cases_DNM.bed | 4,930 | 78

Table: Five WGS  DNM case sample information



### Wong_NC_2016_693_control_SNV- Control sample 

index    | Samples | \# mutations| \# subjects
----|-------|-------|---
1 | Wong_NC_2016_693_control_SNV.bed | 27,092|  693
2 | A->C | 990 (3.65%) | 
3 | A->G | 3,812 (14.07%) |
4 | A->T | 815 (3.01%) |
5 | C->G | 1,274 (4.70%) |
6 | C->T | 5,314 (19.61%) |
7 | C->A | 1,298 (4.79%) |
Table:WGS  DNM control sample information 

```{r, echo=F}
slices <- c(990, 3812, 815, 1274, 5314, 1298) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Wong_NC_2016_693_control_SNV ")
```


There are 585,017,944 C's  and 844,868,045 A's in hg19. 



### Jonsson
De novo WGS of Jonsson data has 1548 controls. 

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C | 3,582
2 | A->G | 13,477  
3 | A->T | 3,331
4 | C->G | 4,822
5 | C->T | 21,220 
6 | C->A | 3,862

```{r, echo=F}
slices <- c(3582, 13477, 3331, 4822, 21220, 3862) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Jonsson data ")
```

### Simons-control


WGS Simons control data

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C | 3,899
2 | A->G | 15,083  
3 | A->T | 3,664
4 | C->G | 5,393
5 | C->T | 23,977 
6 | C->A | 5,404

```{r, echo=F}
slices <- c(3899, 15083, 3664, 5393, 23977, 5404) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Simons control data ")
```

### Yuen data 

WGS Simons control data

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C | 4,239
2 | A->G | 15,893  
3 | A->T | 4,086
4 | C->G | 5,433
5 | C->T | 24,418 
6 | C->A | 5,942

```{r, echo=F}
slices <- c(4239, 15893, 4086, 5433, 24418, 5942) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Yuen data ")
```

### Sample summary

index    | mutation type |  Wong-control| Jonsson | simons-control| Yuen
----|-------|-------|------------|-----------|-
1 | A->C | 990 (3.65%) |  3,582 | 3,899 | 4,239
2 | A->G | 3,812 (14.07%) |13,477 | 15,083 | 15,893
3 | A->T | 815 (3.01%) | 3,331 | 3,664 | 4,086
4 | C->G | 1,274 (4.70%) |4,822 | 5,393 | 5,433
5 | C->T | 5,314 (19.61%) | 21,220 | 23,977 | 24,418
6 | C->A | 1,298 (4.79%) | 3,862 | 5,404 | 5,942


## Data processing and model fitting 

### Data discretization and categorization

* Continuous features are converted  into  binary discrete variables.  
* use R function Split to cluster windows with similar binary (original and converted) features.  

### Model fitting for each mutation type separately 

There are two ways to fit the model, GLM and optims. Both methods give the same esrimate either for whole genome or categorized windows as whown in simulations. Here only present the result by GLM with categorization,   

* GLM (use R function GLM) with offset term
  
  $$y\sim Pois( e^{\mu+X^T\beta})$$

  
  (1) With categorization,  response   $y$ is the ** number of mutations ** from all windows  within a category and covariate feature
  (2) offset term uses 7-mer ERV baseline mutation rate, which is sum of 7mer-mutation rate in a window for each mutation type, e.g. for A->C, sum over all 7-mer mutation rate with A->C in the center.     
  
  (2) intercept is included and the same across all categories
  (3) confidence interval could be obtained as well


## ERV baseline mutation rate 

index  | mutation type |  mutations rate  
----|-------|-------|---
1 | A->C | $1.6627\times 10^{-9}$   
2 | A->G | $6.1805 \times 10^{-9}$
3 | A->T | $1.5633 \times 10^{-9}$
4 | C->G | $2.8075 \times 10^{-9}$
5 | C->T | $1.2885 \times 10^{-8}$
6 | C->A | $3.3977 \times 10^{-9}$
Table: Average mutation rate of different mutation types in ERV paper. 


## Whole genome-fit 14 features simultaneously   

The window length is set 100 bp. Use GLM to fit the model, and stated otherwise. 




```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T", "C_to_A.CpG","C_to_A.nonCpG", "C_to_G.CpG", "C_to_G.nonCpG", "C_to_T.CpG", "C_to_T.nonCpG")
cova=c("DHS", "CpGisland", "Lamin", "GC", "RT", "RR", "Exon", "H3K27ac", "H3K27me3", "H3K36me3", "H3K4me1","H3K4me3","H3K9ac", "H3K9me3")
```


```{r, echo=F}
wong.control.para.est=list(); simons.control.para.est=list(); jonsson.para.est=list()
yuen.para.est=list()
for (i in 1:length(mut.type))
{
  wong.control.para.est[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Wong_NC_2016_693_control_SNV_100bp.window_mutation", mut.type[i], "RData", sep="."))
  simons.control.para.est[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Simons\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
  jonsson.para.est[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Jonsson_Yuen\\comb_SNPs_Jonsson.mutation", mut.type[i], "RData", sep="."))
  yuen.para.est[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Jonsson_Yuen\\comb_SNPs_Yuen.mutation", mut.type[i], "RData", sep="."))
}
names(wong.control.para.est)=mut.type
names(simons.control.para.est)=mut.type
names(jonsson.para.est)=mut.type
names(yuen.para.est)=mut.type
```

```{r, echo=F}
for (cova.index in 1:length(cova))
{
#  cova.index=1
cova.across.mut=matrix(nrow=4, ncol=length(mut.type))

wong.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(wong.control.conf.inter)=c("lower", "upper")
rownames(wong.control.conf.inter)=names(wong.control.para.est)

simons.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter)=c("lower", "upper")
rownames(simons.control.conf.inter)=names(simons.control.para.est)

jonsson.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(jonsson.conf.inter)=c("lower", "upper")
rownames(jonsson.conf.inter)=names(jonsson.para.est)

yuen.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(yuen.conf.inter)=c("lower", "upper")
rownames(yuen.conf.inter)=names(yuen.para.est)

for (i in 1:length(mut.type))
{
  cova.across.mut[1,i]=wong.control.para.est[[i]]$para.est[cova.index+1]
  wong.control.conf.inter[i,]=wong.control.para.est[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[2,i]=simons.control.para.est[[i]]$para.est[cova.index+1]
  simons.control.conf.inter[i,]=simons.control.para.est[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[3,i]=jonsson.para.est[[i]]$para.est[cova.index+1]
  jonsson.conf.inter[i,]=jonsson.para.est[[i]]$para.conf[cova.index+1, ]
  
  cova.across.mut[4,i]=yuen.para.est[[i]]$para.est[cova.index+1]
  yuen.conf.inter[i,]=yuen.para.est[[i]]$para.conf[cova.index+1, ]
  
}
colnames(cova.across.mut)=names(wong.control.para.est)

barcenters=barplot(cova.across.mut, beside=T,ylab=expression(paste(beta)),col=c("red", "blue", "green", "yellow"), legend=c("Wong_contr", "simons_contr", "jonsson", "yuen"), ylim=c(min(wong.control.conf.inter, simons.control.conf.inter, jonsson.conf.inter, yuen.conf.inter, cova.across.mut), max(wong.control.conf.inter, simons.control.conf.inter, jonsson.conf.inter, yuen.conf.inter,cova.across.mut)),  main=cova[cova.index],  cex.names=0.65, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=2)


segments(barcenters, rbind(wong.control.conf.inter[,1], simons.control.conf.inter[,1], jonsson.conf.inter[,1], yuen.conf.inter[,1]), barcenters,rbind(wong.control.conf.inter[,2], simons.control.conf.inter[,2], jonsson.conf.inter[,2], yuen.conf.inter[,2]), lwd = 1.5)

arrows(barcenters, rbind(wong.control.conf.inter[,1], simons.control.conf.inter[,1], jonsson.conf.inter[,1], yuen.conf.inter[,1]), barcenters,rbind(wong.control.conf.inter[,2], simons.control.conf.inter[,2], jonsson.conf.inter[,2], yuen.conf.inter[,2]), lwd = 1.5, angle = 90,code = 3, length = 0.05)
} # end of cova.index
```


* Fit 14 covariate features simultaneously.

* 95% confidence interval by GLM is often big.  



#### Parameter estimate comparison 



```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T")
cova=c("DHS", "CpGisland", "Lamin", "GC", "RT", "RR", "Exon", "H3K27ac", "H3K27me3", "H3K36me3", "H3K4me1","H3K4me3","H3K9ac", "H3K9me3")
```


```{r, echo=F}
simons.control.para.est.100bp=list(); simons.control.para.est.100k=list()
for (i in 1:length(mut.type))
{
  simons.control.para.est.100bp[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Simons\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
 simons.control.para.est.100k[[i]]=loadRData(paste("D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Simons\\100kwindow\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep=".")) 
}
names(simons.control.para.est.100bp)=mut.type
names(simons.control.para.est.100k)=mut.type
```

```{r, echo=F}
for (cova.index in 1:length(cova))
{
#  cova.index=1
cova.across.mut=matrix(nrow=2, ncol=length(mut.type))


simons.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter)=c("lower", "upper")
rownames(simons.control.conf.inter)=names(simons.control.para.est.100bp)

simons.control.conf.inter.100k=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter.100k)=c("lower", "upper")
rownames(simons.control.conf.inter.100k)=names(simons.control.para.est.100k)


for (i in 1:length(mut.type))
{
  
  cova.across.mut[1,i]=simons.control.para.est.100bp[[i]]$para.est[cova.index+1]
  simons.control.conf.inter[i,]=simons.control.para.est.100bp[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[2,i]=simons.control.para.est.100k[[i]]$para.est[cova.index+1]
  simons.control.conf.inter.100k[i,]=simons.control.para.est.100k[[i]]$para.conf[cova.index+1,]
  
  
  
}
colnames(cova.across.mut)=names(simons.control.para.est.100bp)

barcenters=barplot(cova.across.mut, beside=T,ylab=expression(paste(beta)),col=c("red", "blue"), legend=c("simons_contr_100bp", "simons_contr_100k"), ylim=c(min(simons.control.conf.inter, simons.control.conf.inter.100k), max(simons.control.conf.inter, simons.control.conf.inter.100k)),  main=cova[cova.index],  cex.names=0.65, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=2)


segments(barcenters, rbind(simons.control.conf.inter[,1], simons.control.conf.inter.100k[,1]), barcenters,rbind(simons.control.conf.inter[,2], simons.control.conf.inter.100k[,2]), lwd = 1.5)

arrows(barcenters, rbind(simons.control.conf.inter[,1], simons.control.conf.inter.100k[,1]), barcenters,rbind(simons.control.conf.inter[,2], simons.control.conf.inter.100k[,2]), lwd = 1.5, angle = 90,code = 3, length = 0.05)
} # end of cova.index
```


## Fitting diagnostic-Simons-control


### Standardized residual


#### Simons-control 

* 100 bp windows 


index  | mutation type | # total windows |# window $p.val<0.05$ |  # window $p.val=0$
----|-------|-------|---|-----------
1 | A->C | 26,839,568 | 3,899 |   3,870
2 | A->G | 15,067 | 15,067 | 10,806
3 | A->T | 26,839,598 | 3,656 | 3,649




* 100k windows 



```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T", "C_to_A", "C_to_G", "C_to_T")
```



```{r, echo=F}
path="D:\\ResearchWork\\StatisticalGenetics\\Mutation-Rate-project\\parameter_estimate\\Simons\\100kwindow\\"
prefix="table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation."
for (i in 1:length(mut.type))
{
simons_control_100k=read.table(paste(path, prefix, mut.type[i], ".ExpObsVar.bed", sep=""), header=F)
simons_control_100k=simons_control_100k[complete.cases(simons_control_100k), ]
y.min=min(simons_control_100k[,6])-1; y.max=max(simons_control_100k[,6])+1

op <- par(mar = c(5,6,4,2) + 0.1)
plot(simons_control_100k[,6], type="l", ylim=c(y.min, y.max), ylab="", xlab="window index", main=mut.type[i])
abline(h=1.75, col="red")
#axis(2,cex.axis=1.2)
mtext(expression(frac(y[i]-hat(mu)[i], sqrt(hat(mu)[i]))), side=2, line=2.2, cex=1.2)
par(op)
}


```

* use parameter estimated for 100kb windows to calculate residuals and estimate from 100 bp windows are not applicable because covariates have changed. 

* shown are only windows with numercial residuals and windows with both zero observed and expected mutation leading to NA residuals are removed. So neighbor windows in this plot may not be phisically close in chromosome.  

* View via IGV 

![ Residuals of 100kb windows with p value less than 0.05  in whole genome ](figure/Simons_100kWindow.png)



![ Residuals of 100kb windows with p values less than 0.05 in large chunks ](figure/Simons_100kWindow_largechunk.png)




## Cross validation 



## Session information


<!-- Insert the session information into the document -->
```{r session-info}
```
