---
title: "De novo mutations of ASD"
author: "Shengtong  Han"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
library(knitr)
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(ape)
library(tidyverse)
set.seed(123)
```


```{r, echo=F}
loadRData <- function(fileName){
#loads an RData file, and returns it
    load(fileName)
    get(ls()[ls() != "fileName"])
}
```


## Samples



### Case Samples 



index    | Samples | \# mutations| \# subjects
----|-------|-------|---
1 | Jiang_cases_DNM.bed | 2,091| 32
2 | Michaelson_cases_DNM.bed |581 | 10
3 | Yuen_NM2015_cases_DNM.bed | 9,381| 162
4 | Wu_cases_DNM.bed | 1,915 | 32
5 | Kong_cases_DNM.bed | 4,930 | 78

Table: Five WGS  DNM case sample information

* Jiang_cases_DNM.bed [Detection of clinically relevant genetic variants in autism spectrum disorder by whole-genome sequencing](https://pubmed.ncbi.nlm.nih.gov/23849776/), AJHG 2013. 

* Michaelson_cases_DNM.bed [Whole-genome sequencing in autism identifies hot spots for de novo germline mutation](https://pubmed.ncbi.nlm.nih.gov/23260136/), cell, 2012

* Yuen_NM2015_cases_DNM.bed [Whole-genome sequencing of quartet families with autism spectrum disorder](https://pubmed.ncbi.nlm.nih.gov/25621899/), nature medicine, 2015

* Wu_cases_DNM.bed [unpublished]

* Kong_cases_DNM.bed [Rate of de novo mutations and the importance of father's age to disease risk](https://www.nature.com/articles/nature11396), nature, 2012



### Wong_NC_2016_693_control_SNV- Control sample 

index    | Samples | \# mutations| \# subjects
----|-------|-------|---
1 | Wong_NC_2016_693_control_SNV.bed | 27,092|  693
2 | A->C | 990 (3.65%) | 
3 | A->G | 3,812 (14.07%) |
4 | A->T | 815 (3.01%) |
5 | C->G | 1,274 (4.70%) |
6 | C->T | 5,314 (19.61%) |
7 | C->A | 1,298 (4.79%) |
Table:WGS  DNM control sample information 

```{r, echo=F}
slices <- c(990, 3812, 815, 1274, 5314, 1298) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Wong_NC_2016_693_control_SNV ")
```


There are 585,017,944 C's  and 844,868,045 A's in hg19. 



### Jonsson

De novo WGS of Jonsson data has 1548 controls. 

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C | 3,582
2 | A->G | 13,477  
3 | A->T | 3,331
4 | C->G | 4,822
5 | C->T | 21,220 
6 | C->A | 3,862

```{r, echo=F}
slices <- c(3582, 13477, 3331, 4822, 21220, 3862) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Jonsson data ")
```

### Simons-control


WGS Simons control data

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C (+ T->G) | 3,899+3,900=7,799
2 | A->G (+ T->C) | 15,083+14,919=30,002  
3 | A->T (+ T->A) | 3,664+3,743=7,407
4 | C->G (+ G->C) | 5,393+5,308=10,701
5 | C->T (+ G->A) | 23,977+23,907=47,884 
6 | C->A (+ G->T)| 5,404+5,652=11,056

```{r, echo=F}
slices <- c(7799, 30002, 7404, 10701, 47884, 11056) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Simons control data ")
```

### Yuen data 

Yuen data 

index    | mutation type | \# mutations
----|-------|-------|---
1 | A->C | 4,239
2 | A->G | 15,893  
3 | A->T | 4,086
4 | C->G | 5,433
5 | C->T | 24,418 
6 | C->A | 5,942

```{r, echo=F}
slices <- c(4239, 15893, 4086, 5433, 24418, 5942) 
lbls <- c("A->C", "A->G", "A->T", "C->G", "C->T", "C->A")
pct <- round(slices/sum(slices)*100)
lbls <- paste(lbls, pct) # add percents to labels 
lbls <- paste(lbls,"%",sep="") # ad % to labels 
pie(slices,labels = lbls, col=rainbow(length(lbls)),
  	main="Percentage of mutation types in Yuen data ")
```

### Sample summary

index    | mutation type |  Wong-control| Jonsson | simons-control| Yuen
----|-------|-------|------------|-----------|-
1 | A->C | 990 (3.65%) |  3,582 | 3,899 | 4,239
2 | A->G | 3,812 (14.07%) |13,477 | 15,083 | 15,893
3 | A->T | 815 (3.01%) | 3,331 | 3,664 | 4,086
4 | C->G | 1,274 (4.70%) |4,822 | 5,393 | 5,433
5 | C->T | 5,314 (19.61%) | 21,220 | 23,977 | 24,418
6 | C->A | 1,298 (4.79%) | 3,862 | 5,404 | 5,942

```{r, echo=T, eval=F}
Wong data: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Wong_NC_2016_693_control_SNV_with_allele_info.txt
Jonson: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Jonsson_Yuen2017/comb_SNPs_Jonsson
Yuen:  /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Jonsson_Yuen2017/comb_SNPs_Yuen
Simons: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Simons_WGS
```


* Wong data: [New observations on maternal age effect on germline de novo mutation](https://www.nature.com/articles/ncomms10486), nature communications, 2016

* Jonson, [Whole genome characterization of sequence diversity of 15,220 Icelanders](https://www.nature.com/articles/sdata2017115), Scientific Data, 2017

* Simons: [Genomic Patterns of De Novo Mutation in Simplex Autism](https://pubmed.ncbi.nlm.nih.gov/28965761/), cell, 2017

```{r, echo=T, eval=F}
Folder: /project2/xinhe/Shengtong/mutation_rate/mutation.rate/Simons_WGS/controls
Data for 100bp window: table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.ExpObsVar.bed.bed.add.100bp
Each row shows data of one 100bp window. Each column: chromosome, start, end, observed DNM count in the window, and fitted mutation rate in the window. Those counts are obtained by combing all mutation types together

chr1 	 0 	 100 0 	 0
chr1 	 100 	 200 0 	 0
chr1 	 200 	 300 0 	 0
chr1 	 300 	 400 0 	 0
chr1 	 400 	 500 0 	 0
chr1 	 500 	 600 0 	 0
chr1 	 600 	 700 0 	 0
chr1 	 700 	 800 0 	 0
chr1 	 800 	 900 0 	 0
chr1 	 900 	 1000 0 	 0

```


#### mutaton count by mutationt type 




## Data processing and model fitting 

### Data discretization and categorization

* Continuous features are converted  into  binary discrete variables.  
* use R function Split to cluster windows with similar binary (original and converted) features.  

### Model fitting for each mutation type separately 

There are two ways to fit the model, GLM and optims. Both methods give the same esrimate either for whole genome or categorized windows as whown in simulations. Here only present the result by GLM with categorization,   

* GLM (use R function GLM) with offset term
  
  $$y\sim Pois( e^{\mu+X^T\beta})$$

  
  (1) With categorization,  response   $y$ is the **total number of mutations** from all windows  within a category and covariate feature
  (2) offset term uses 7-mer ERV baseline mutation rate, which is sum of 7-mer mutation rate in a window for each mutation type, e.g. for A->C, sum over all 7-mer mutation rate with A->C in the center.     
  
  (2) intercept is included and the same across all categories
  (3) confidence interval could be obtained as well


## ERV baseline mutation rate 

index  | mutation type |  mutations rate  
----|-------|-------|---
1 | A->C | $1.6627\times 10^{-9}$   
2 | A->G | $6.1805 \times 10^{-9}$
3 | A->T | $1.5633 \times 10^{-9}$
4 | C->G | $2.8075 \times 10^{-9}$
5 | C->T | $1.2885 \times 10^{-8}$
6 | C->A | $3.3977 \times 10^{-9}$
Table: Average mutation rate of different mutation types in ERV paper. 


## Whole genome-fit 14 features simultaneously   

The window length is set 100 bp. Use GLM to fit the model, and stated otherwise. 




```{r, echo=F}
setwd("C:\\Shengtong\\Research\\mutation-rate\\mutation-rate-modeling\\analysis\\")
mut.type=c("A_to_C", "A_to_G", "A_to_T", "C_to_A.CpG","C_to_A.nonCpG", "C_to_G.CpG", "C_to_G.nonCpG", "C_to_T.CpG", "C_to_T.nonCpG")
cova=c("DHS", "CpGisland", "Lamin", "GC", "RT", "RR", "Exon", "H3K27ac", "H3K27me3", "H3K36me3", "H3K4me1","H3K4me3","H3K9ac", "H3K9me3")
```


```{r, echo=F}
wong.control.para.est=list(); simons.control.para.est=list(); jonsson.para.est=list()
yuen.para.est=list()
for (i in 1:length(mut.type))
{
  wong.control.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Wong_NC_2016_693_control_SNV_100bp.window_mutation", mut.type[i], "RData", sep="."))
  simons.control.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
  jonsson.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Jonsson_Yuen\\comb_SNPs_Jonsson.mutation", mut.type[i], "RData", sep="."))
  yuen.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Jonsson_Yuen\\comb_SNPs_Yuen.mutation", mut.type[i], "RData", sep="."))
}
names(wong.control.para.est)=mut.type
names(simons.control.para.est)=mut.type
names(jonsson.para.est)=mut.type
names(yuen.para.est)=mut.type
```

```{r, echo=F}
for (cova.index in 1:length(cova))
{
#  cova.index=1
cova.across.mut=matrix(nrow=4, ncol=length(mut.type))

wong.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(wong.control.conf.inter)=c("lower", "upper")
rownames(wong.control.conf.inter)=names(wong.control.para.est)

simons.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter)=c("lower", "upper")
rownames(simons.control.conf.inter)=names(simons.control.para.est)

jonsson.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(jonsson.conf.inter)=c("lower", "upper")
rownames(jonsson.conf.inter)=names(jonsson.para.est)

yuen.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(yuen.conf.inter)=c("lower", "upper")
rownames(yuen.conf.inter)=names(yuen.para.est)

for (i in 1:length(mut.type))
{
  cova.across.mut[1,i]=wong.control.para.est[[i]]$para.est[cova.index+1]
  wong.control.conf.inter[i,]=wong.control.para.est[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[2,i]=simons.control.para.est[[i]]$para.est[cova.index+1]
  simons.control.conf.inter[i,]=simons.control.para.est[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[3,i]=jonsson.para.est[[i]]$para.est[cova.index+1]
  jonsson.conf.inter[i,]=jonsson.para.est[[i]]$para.conf[cova.index+1, ]
  
  cova.across.mut[4,i]=yuen.para.est[[i]]$para.est[cova.index+1]
  yuen.conf.inter[i,]=yuen.para.est[[i]]$para.conf[cova.index+1, ]
  
}
colnames(cova.across.mut)=names(wong.control.para.est)

barcenters=barplot(cova.across.mut, beside=T,ylab=expression(paste(beta)),col=c("red", "blue", "green", "yellow"), legend=c("Wong_contr", "simons_contr", "jonsson", "yuen"), ylim=c(min(wong.control.conf.inter, simons.control.conf.inter, jonsson.conf.inter, yuen.conf.inter, cova.across.mut), max(wong.control.conf.inter, simons.control.conf.inter, jonsson.conf.inter, yuen.conf.inter,cova.across.mut)),  main=cova[cova.index],  cex.names=0.65, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=2)


segments(barcenters, rbind(wong.control.conf.inter[,1], simons.control.conf.inter[,1], jonsson.conf.inter[,1], yuen.conf.inter[,1]), barcenters,rbind(wong.control.conf.inter[,2], simons.control.conf.inter[,2], jonsson.conf.inter[,2], yuen.conf.inter[,2]), lwd = 1.5)

arrows(barcenters, rbind(wong.control.conf.inter[,1], simons.control.conf.inter[,1], jonsson.conf.inter[,1], yuen.conf.inter[,1]), barcenters,rbind(wong.control.conf.inter[,2], simons.control.conf.inter[,2], jonsson.conf.inter[,2], yuen.conf.inter[,2]), lwd = 1.5, angle = 90,code = 3, length = 0.05)
} # end of cova.index
```


* Fit 14 covariate features simultaneously.

* 95% confidence interval by GLM is often big.  



#### Parameter estimate comparison 



```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T")
cova=c("DHS", "CpGisland", "Lamin", "GC", "RT", "RR", "Exon", "H3K27ac", "H3K27me3", "H3K36me3", "H3K4me1","H3K4me3","H3K9ac", "H3K9me3")
```


```{r, echo=F}
simons.control.para.est.100bp=list(); simons.control.para.est.100k=list()
simons.control.para.est.10k=list()
for (i in 1:length(mut.type))
{
  simons.control.para.est.100bp[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
 simons.control.para.est.100k[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\100kwindow\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
 simons.control.para.est.10k[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\10kwindow\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "RData", sep="."))
 
}
names(simons.control.para.est.100bp)=mut.type
names(simons.control.para.est.100k)=mut.type
names(simons.control.para.est.10k)=mut.type
```

```{r, echo=F}
for (cova.index in 1:length(cova))
{
#  cova.index=1
cova.across.mut=matrix(nrow=3, ncol=length(mut.type))


simons.control.conf.inter=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter)=c("lower", "upper")
rownames(simons.control.conf.inter)=names(simons.control.para.est.100bp)

simons.control.conf.inter.100k=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter.100k)=c("lower", "upper")
rownames(simons.control.conf.inter.100k)=names(simons.control.para.est.100k)

simons.control.conf.inter.10k=matrix(nrow=length(mut.type), ncol=2)
colnames(simons.control.conf.inter.10k)=c("lower", "upper")
rownames(simons.control.conf.inter.10k)=names(simons.control.para.est.10k)


for (i in 1:length(mut.type))
{
  
  cova.across.mut[1,i]=simons.control.para.est.100bp[[i]]$para.est[cova.index+1]
  simons.control.conf.inter[i,]=simons.control.para.est.100bp[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[3,i]=simons.control.para.est.100k[[i]]$para.est[cova.index+1]
  simons.control.conf.inter.100k[i,]=simons.control.para.est.100k[[i]]$para.conf[cova.index+1,]
  
  cova.across.mut[2,i]=simons.control.para.est.10k[[i]]$para.est[cova.index+1]
  simons.control.conf.inter.10k[i,]=simons.control.para.est.10k[[i]]$para.conf[cova.index+1,]
  
}
colnames(cova.across.mut)=names(simons.control.para.est.100bp)

barcenters=barplot(cova.across.mut, beside=T,ylab=expression(paste(beta)),col=c("red", "blue", "green"), legend=c("simons_contr_100bp", "simons_contr_10k", "simons_contr_100k"), ylim=c(min(simons.control.conf.inter, simons.control.conf.inter.100k, simons.control.conf.inter.10k), max(simons.control.conf.inter, simons.control.conf.inter.100k, simons.control.conf.inter.10k)),  main=cova[cova.index],  cex.names=0.65, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=2)


segments(barcenters, rbind(simons.control.conf.inter[,1], simons.control.conf.inter.10k[,1], simons.control.conf.inter.100k[,1]), barcenters,rbind(simons.control.conf.inter[,2], simons.control.conf.inter.10k[,2], simons.control.conf.inter.100k[,2]), lwd = 1.5)

arrows(barcenters, rbind(simons.control.conf.inter[,1], simons.control.conf.inter.10k[,1], simons.control.conf.inter.100k[,1]), barcenters,rbind(simons.control.conf.inter[,2], simons.control.conf.inter.10k[,2], simons.control.conf.inter.100k[,2]), lwd = 1.5, angle = 90,code = 3, length = 0.05)
} # end of cova.index
```



## Parameter estimate with variable selection



```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T", "C_to_A.CpG","C_to_A.nonCpG", "C_to_G.CpG", "C_to_G.nonCpG", "C_to_T.CpG", "C_to_T.nonCpG")
cova=c("DHS", "CpGisland", "Lamin", "GC", "RT", "RR", "Exon", "H3K27ac", "H3K27me3", "H3K36me3", "H3K4me1","H3K4me3","H3K9ac", "H3K9me3")
```

```{r, echo=F}
simons.control.para.est=list()
for (i in 1:length(mut.type))
{
  simons.control.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\100bpwindow_varselection\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "var.selection.RData", sep="."))
}
```


```{r, echo=F}
for (mut in 1:length(mut.type))
{
#  mut=1
pick.cova=cova[simons.control.para.est[[mut]]$pick.cova.index]
cova.fea=matrix(nrow=1, ncol=length(pick.cova))


simons.control.conf.inter=matrix(nrow=length(pick.cova), ncol=2)
colnames(simons.control.conf.inter)=c("lower", "upper")
rownames(simons.control.conf.inter)=pick.cova


for (i in 1:length(pick.cova))
{
  
  cova.fea[1,i]=simons.control.para.est[[mut]]$para.est[i+1]
  simons.control.conf.inter[i,]=simons.control.para.est[[mut]]$para.conf[i+1,]
  
  
}
colnames(cova.fea)=pick.cova

barcenters=barplot(cova.fea, beside=T,ylab=expression(paste(beta)),col=c("red"), legend=c("simons_contr_100bp"), ylim=c(min(simons.control.conf.inter), max(simons.control.conf.inter)),  main=mut.type[mut],  cex.names=0.65, args.legend = list(x ='topright', bty='n', inset=c(-0.005,0), cex=0.8), xpd = FALSE, las=2)


segments(barcenters, simons.control.conf.inter[,1],  barcenters,simons.control.conf.inter[,2], lwd = 1.5)

arrows(barcenters, simons.control.conf.inter[,1], barcenters,simons.control.conf.inter[,2], lwd = 1.5, angle = 90,code = 3, length = 0.05)
} # end of mut
```


* compute the ratio of confidence interval over absolute value of estimated coefficient, rank them, and  choose maximum of the top 25% quantile  or 1. That is at least there are 25% covariate features with smallest ratios are selected as significant features.  


## methylation feature only 

```{r, echo=F}
mut.type=c("A_to_C", "A_to_G", "A_to_T", "C_to_A.CpG","C_to_A.nonCpG", "C_to_G.CpG", "C_to_G.nonCpG", "C_to_T.CpG", "C_to_T.nonCpG")
complement.mut.type=c("T_to_G", "T_to_C", "T_to_A", "G_to_T.CpG","G_to_T.nonCpG", "G_to_C.CpG", "G_to_C.nonCpG", "G_to_A.CpG", "G_to_A.nonCpG")
cova=c("15Brain_5W_embryo1_methy")
```



```{r, echo=F, eval=T}
path="C:\\Shengtong\\Research\\mutation-rate\\parameter_estimate\\Simons\\100bpwindow\\MethyFeatureOnly\\"

#setwd("C:/Shengtong/Research/mutation-rate/mutation-rate-modeling/analysis") ### set working directory 

########### don't know why it doesn't work 
simons.control.para.est=list()
#for (i in 1:9)
#{
#  simons.control.para.est[[i]]=loadRData(paste("..\\..\\parameter_estimate\\Simons\\100bpwindow\\MethyFeatureOnly\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "and.mutation", complement.mut.type[i],  "100bpwindow.with.standardize.covariate.methyONLY.RData", sep="."))
  
#simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
#}

i=1
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 i=2
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 i=3
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 i=4
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 #i=5
 #simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 # simons.control.para.est[[i]]=loadRData("C:\\Shengtong\\Research\\mutation-rate\\parameter_estimate\\Simons\\100bpwindow\\MethyFeatureOnly\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.C_to_A.nonCpG.and.mutation.G_to_T.nonCpG.100bpwindow.with.standardize.covariate.methyONLY.RData")
 i=6
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
# i=7
# simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
 i=8
 simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))
# i=9
# simons.control.para.est[[i]]=loadRData(paste(path, "table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.methyONLY.RData", sep=""))

```

* base level methylation are averaged in a window as window level methylation, then converted into binary variable with mean as threshold to fit the model 
* regress mutation count on methylation feature only 
* parameter estimated ($\hat{\beta}$) + 95% CI (vertical line)
* methylation samples are downloaded from [GSE63818](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63818) 
* methylation samples are strand specific. Plus strand data uses plus strand methylation sample and the same for minus strand. 
* methylation: GSE63818_Brain_5W_embryo1_methylation


```{r, echo=F}
methy.fea=matrix(nrow=length(mut.type), ncol=3)
sub_set=c(1,2,3,4,6,8)
for (i in 1:length(sub_set))
methy.fea[sub_set[i],]=c(simons.control.para.est[[sub_set[i]]]$para.est[2],simons.control.para.est[[sub_set[i]]]$para.conf[2,]) 

methy.fea[5,]=c(-0.005428626, -0.05213374,  0.04088773) # manually add this numbers because those files couldn't be read into 
methy.fea[7,]=c(0.01748771, -0.02938896, 0.06397908)
methy.fea[9,]=c(0.03527426, 0.007650857, 0.06276593)
methy_only=tibble(mutation_type=mut.type, estimate=methy.fea[,1], lower_bound=methy.fea[,2], upper_bound=methy.fea[,3])

ggplot(methy_only, aes(x=mutation_type, y=estimate, fill=mutation_type)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  theme(legend.position = "none")+ # remove legend 
  geom_errorbar(aes(ymin=lower_bound, ymax=upper_bound), width=.2,
                 position=position_dodge(.9))+
  ylab(expression(paste(hat(beta))))+
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=8))+
  ggtitle("Brain_5W_embryo1_methy")+
  theme(plot.title = element_text(hjust = 0.5, size=15))  #center the title
```


```{r, echo=F, message=FALSE, warning=FALSE}
path="C:\\Shengtong\\Research\\mutation-rate\\parameter_estimate\\Simons\\100bpwindow\\MethyFeatureOnly\\"
#methy=c("15GSE63818_PGC_7W_embryo1_M_methylation_calling.bed",
#  "15GSE63818_PGC_10W_embryo1_F_rep1_methylation_calling.bed", #"15GSE63818_PGC_10W_embryo1_M_methylation_calling.bed", #"15GSE63818_PGC_11W_embryo1_F_methylation_calling.bed", #"15GSE63818_PGC_11W_embryo1_M_rep1_methylation_calling.bed", #"15GSE63818_PGC_13W_embryo1_M_methylation_calling.bed",  #"15GSE63818_Soma_17W_embryo1_F_methylation_calling.bed", #"15GSE63818_Soma_19W_embryo1_M_methylation_calling.bed")

methy=c("PGC_7W_embryo1_M",
  "PGC_10W_embryo1_F", "PGC_10W_embryo1_M", "PGC_11W_embryo1_F", "PGC_11W_embryo1_M", "PGC_13W_embryo1_M",  "Soma_17W_embryo1_F", "Soma_19W_embryo1_M")

#setwd("C:\\Shengtong\\Research\\mutation-rate\\mutation-rate-modeling\\analysis\\") ### set working directory 
###############################################
####### manually type the results 
para_est=numeric(length(mut.type)*length(methy)); lower_bound=numeric(length(mut.type)*length(methy)); upper_bound=numeric(length(mut.type)*length(methy))
para_est_mut=data.frame(para_est=para_est, lower_bound=lower_bound, upper_bound=upper_bound, mut_type=rep(mut.type, each=length(methy)), methy=rep(methy, length(mut.type)))
para_est_mut[1,1:3]=c(0.05894306, 0.006037771, 0.1113505)  # A->C
para_est_mut[2,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[3,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[4,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[5,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[6,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[7,1:3]=c(0.05894306, 0.006037771, 0.1113505)
para_est_mut[8,1:3]=c(0.0571056, 0.004139021, 0.1095724)

para_est_mut[9,1:3]=c(0.03823484, 0.0111415, 0.06519488) # A->G
para_est_mut[10,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[11,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[12,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[13,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[14,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[15,1:3]=c(0.03823484, 0.0111415, 0.06519488)
para_est_mut[16,1:3]=c(0.03823484, 0.0111415, 0.06519488)

para_est_mut[17,1:3]=c(0.03284713, -0.02186639, 0.08701867) # A->T
para_est_mut[18,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[19,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[20,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[21,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[22,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[23,1:3]=c(0.03284713, -0.02186639, 0.08701867)
para_est_mut[24,1:3]=c(0.03284713, -0.02186639, 0.08701867)

para_est_mut[25,1:3]=c(0.07394379, -0.06825234, 0.2127826) # C->A  CpG
para_est_mut[26,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[27,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[28,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[29,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[30,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[31,1:3]=c(0.07394379, -0.06825234, 0.2127826)
para_est_mut[32,1:3]=c(0.07394379, -0.06825234, 0.2127826)

para_est_mut[33,1:3]=c(-0.005428626, -0.05213374, 0.04088773) # C->A  nonCpG
para_est_mut[34,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[35,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[36,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[37,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[38,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[39,1:3]=c(-0.005428626, -0.05213374, 0.04088773)
para_est_mut[40,1:3]=c(-0.005428626, -0.05213374, 0.04088773)

para_est_mut[41,1:3]=c(0.06348866, -0.09086077, 0.2138658) # C->G  CpG
para_est_mut[42,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[43,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[44,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[45,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[46,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[47,1:3]=c(0.06348866, -0.09086077, 0.2138658)
para_est_mut[48,1:3]=c(0.06348866, -0.09086077, 0.2138658)

para_est_mut[49,1:3]=c(0.01748771, -0.02938896, 0.06397908) # C->G  nonCpG
para_est_mut[50,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[51,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[52,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[53,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[54,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[55,1:3]=c(0.01748771, -0.02938896, 0.06397908)
para_est_mut[56,1:3]=c(0.01748771, -0.02938896, 0.06397908)

para_est_mut[57,1:3]=c(0.08481473, 0.05282837, 0.1166308) # C->T  CpG
para_est_mut[58,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[59,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[60,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[61,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[62,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[63,1:3]=c(0.08481473, 0.05282837, 0.1166308)
para_est_mut[64,1:3]=c(0.08481473, 0.05282837, 0.1166308)

para_est_mut[65,1:3]=c(0.03527426, 0.007650857, 0.06276593) # C->T  nonCpG
para_est_mut[66,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[67,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[68,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[69,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[70,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[71,1:3]=c(0.03527426, 0.007650857, 0.06276593)
para_est_mut[72,1:3]=c(0.03527426, 0.007650857, 0.06276593)
###############################################
########### don't know why it doesn't work
#simons.control.para.est=list()
#for (j in 1:length(methy))
#{
 # simons.control.para.est[[i]]=load.Rdata2(paste("table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation", mut.type[i], "and.mutation", complement.mut.type[i],  "100bpwindow.with.standardize.covariate.15GSE63818_PGC_10W_embryo1_F_rep1_methylation_calling.bed.RData", sep="."))
  
#simons.control.para.est[[i]]=loadRData(paste( "..\\..\\parameter_estimate\\Simons\\100bpwindow\\MethyFeatureOnly\\table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate.15GSE63818_PGC_10W_embryo1_F_rep1_methylation_calling.bed.RData", sep=""))
  
 # simons.control.para.est[[j]]=read.table(paste(path,"table.ASCWGS_20180504.WGS1902_hg19_controls_SNV_remove_recurrent_mutation.", mut.type[i],".and.mutation.",complement.mut.type[i],".100bpwindow.with.standardize.covariate",methy[j], ".txt", sep=""))
#}


#para_est_mut=para_est_mut %>% rename("15GSE63818_PGC_7W_embryo1_M_methylation_calling.bed"="7W_embryo1_M")
  
ggplot(para_est_mut, aes(x=methy, y=para_est, fill=methy)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  theme(legend.position = "none")+ # remove legend 
  geom_errorbar(aes(ymin=lower_bound, ymax=upper_bound), width=.2,
                 position=position_dodge(.9))+
  ylab(expression(paste(hat(beta))))+
  theme(axis.text.x = element_text(angle = 60, vjust = 0.5, size=5))+
  ggtitle("")+
  theme(plot.title = element_text(hjust = 0.5, size=15))+  #center the title
facet_wrap(~mut_type)
```



* For each mutation type, $\widehat{\beta}$ is nearly identical across difference methylation sample. 


## Session information


<!-- Insert the session information into the document -->
```{r session-info}
```
