---
title: "Genomic features"
author: "Shengtong Han"
output: html_document
---



```{r load packages, results='hide', include=FALSE}
library(RSQLite)
library(dplyr)
library(knitr)
library(git2r)
library(tidyverse)
```


## Genomic features 

These features are from the [reference data][data]

[data]:https://github.com/carjed/smaug-genetics


 11 binary features 

* EXON: (416,549)
* H3K4me1
* H3K4me3
* H3K9ac
* H3K9me3
* H3K27ac
* H3K27me3
* H3K36me3
* [CpGI][cpgi] (~1kb, 65,699 windows in total)

[cpgi]:http://web.stanford.edu/class/bios221/data/model-based-cpg-islands-hg19.txt

* LAMIN (300,000, 1301 windows in total)
* DHS (DNase  hypersensitive sites, 1~200 bp window, 1867665 windows in total)

3 continuous features:

* RR [recombination rate](https://www.nature.com/scitable/definition/recombination-226/) 
* TIME (Replication timing, 2,386,623 sites) Regions with negative values are later replicating regions. 
* GC 

Others: 

* Human Accelerated Regions (HAR, 562 windows)

## Recombination rate 

* recombination rate in 100 bp windows 


```{r, echo=F, warning=F, message=F, cache=T}
path="C:\\Shengtong\\Research\\mutation-rate\\CovariateFeature\\"
recombination_rate=read.table(paste(path, "feature.6RecombRate.bed",sep=""), header=F)
summary(recombination_rate$V4)
hist(recombination_rate$V4, xlab="recombination rate", main="")
# plot(seq(1, nrow(recombination_rate)), recombination_rate$V4, xlab="window index", ylab="recombination rate")
```
![image info](C:/Shengtong/Research/mutation-rate/CovariateFeature/RecombinationRate.jpeg)

```{r, echo=F}
group=c("(A):<=1", "(B):(1,10)", "(C):(10,100)", "(D):>100")
count=numeric()
count[1]=nrow(recombination_rate[recombination_rate$V4<1,])
count[2]=nrow(recombination_rate[recombination_rate$V4>1 & recombination_rate$V4<10,])
count[3]=nrow(recombination_rate[recombination_rate$V4>10 & recombination_rate$V4<100,])
count[4]=nrow(recombination_rate[recombination_rate$V4>100,])

ggplot(data=data.frame(mutation_rate=group, count=count), aes(x=mutation_rate, y=count, fill=mutation_rate)) +
  geom_bar(stat="identity")+
  xlab("recombination rate")+ylab("window counts")

data.frame(group=group, count=count)
```

### chr8 

```{r, echo=F, warning=F, message=F, cache=T}
recombination_rate_chr8=recombination_rate[recombination_rate$V1=="chr8",]
summary(recombination_rate_chr8$V4)
hist(recombination_rate_chr8$V4, xlab="recombination rate", main="chr8")
plot(seq(1, nrow(recombination_rate_chr8)), recombination_rate_chr8$V4, xlab="window index", ylab="recombination rate", main="chr8")
```

## DNA methylation 

Methylation data are downloaded at GEO [GSE63818](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63818) from paper [The Transcriptome and DNA Methylome Landscapes of Human Primordial Germ Cells, Cell, 2015](https://www.sciencedirect.com/science/article/pii/S0092867415005632)
