---
title: "Statistical Modeling"
author: "Shengtong Han"
date: 
output: html_document
---



## Base level


At every genomic locus, observed number of mutations, $y_i$ is modeled as Poisson distribution as 

$y_i \sim Pois(\mu_i)$

There are several covariates affecting the mutation rate, such as tri-nucleotide context, parents' age, etc. To relate the mutation rate to covariates $X_i$, Poisson regression could be one way to do that 

$log(E(y_i|X_i))=log(\mu_i)=X_i^T\beta$

Thus 

$\mu_i=e^{X_i^T\beta}$. 


The likelihood function is 

$$
\begin{aligned} 
L(\beta|Y)&=\prod_i\frac{[e^{X_i^T\beta}]^{Y_i}e^{-e^{X_i^T\beta}}}{Y_i!}\\
&=e^{\sum_i[X_i^T\beta Y_i-e^{X_i^T\beta}]} \times constant
\end{aligned}
$$

The log likelihood is 

$$
\begin{aligned}
l(\beta)&=\sum_i[X_i^T\beta Y_i-e^{X_i^T\beta}]+constant\\
&=\sum_i(X_i^T\beta)Y_i-\sum_ie^{X_i^T\beta}+constant
\end{aligned}
$$



## Region level

It's computationally hard to infer the mutation rate at single base since there are 3B bases in human genome. So consider the window level. Within  a window, collapse all bases together, denote $Y$ by the total number of mutations, assume $X_i$ are the same across all bases. Within a window, assuming $Y_i\sim Pois(\mu_i)$ are independent poisson random variables, the sum $Y=\sum_iY_i \sim Pois (\sum_i\mu_i)$


Considering baseline mutation rate $\mu_{i0}$ at every base $i$, then $Y \sim Pois(\sum_i \mu_{i0} \mu_i)$, where $\sum_i \mu_{i0}\mu_i=\sum_i\mu_{i0}e^{X_i^T\beta}=e^{X^T\beta}\sum_i\mu_{i0}$, assuming $x_i^T\beta$ are the same across all bases in the window and $\sum_i \mu_{i0}$ is the total baseline mutation for the window. Then $Y_j \sim Pois(\mu_{j0}e^{X_j^T\beta})$


The likelihood function is 


\begin{equation}
L(\beta|Y)=\prod_j\frac{[\mu_{j0}e^{X_j^T\beta}]^{Y_j}e^{-\mu_{j0}e^{X_j^T\beta}}}{Y_j!}
\end{equation}


and

* $Y_j$; total number of mutations in  window $j$
* $X_j$: a vector of covariates for the window, assuming the same across all bases in the window





And the log likelihood function is 


\begin{equation}
\tag{1}
l(\beta)=\sum_j[Y_jlog(\underbrace{\mu_{j0}e^{X_j^T\beta}}_{\mu_j})]-\sum_j[\underbrace{\mu_{j0}e^{X_j^T\beta}}_{\mu_j}]+constant\\
=\sum[Y_jlog(\mu_j)]-\sum_j[\mu_j]+constant
\end{equation}






## Categorization

Human genome has 3B bases, even for 100 bp windows, the computation is still expensive. To speed up, one way is to put all windows with similar features into one category and treat the whole category as a single unit. Suppose we have $P$ binary features (continuous features could be categorized into discrete values). Totally it is going to be $2^P$ feature categories.  Usually $2^P << \frac{3B}{L}$, $L$ is window length. 

Let $c$ index category. $Y_c$ is the total number of mutations, $\mu_{c0}$ is the baseline mutation rate, $X_c$ is the feature covariates. 


The likelihood function is 

$$
\begin{aligned}
 logP(Y|\beta) & = \sum_j[Y_jlog(\mu_{j0}e^{X_j^T\beta})]-\sum_j[\mu_{j0}e^{X_j^T\beta}]+constant(following ~LH~of~region~level)\\
 &=\sum_c\{Y_c(X_c^T\beta)-\mu_{c0}e^{X_c^T\beta} \}+constant
\end{aligned}
$$

* $\mu_{c0}=\sum_j\mu_{j0}$ in category $c$.
* The last equation is because for category $c$, all $X_j=X_c$ are the same, so $\sum_jY_j(X_j^T\beta)=X_c^T\beta \sum_jY_j=Y_c(X_c^T\beta)$. 

Thus $Y_c\sim Pois(\mu_{c0}e^{X_c^T\beta})$. Another way to understand it is that since $Y_j \sim Pois(\mu_{j0}e^{X_j^T\beta})$, in category $c$, the sum of $\sum_j Y_j \sim Pois(\sum_j \mu_{j0}e^{X_j^T\beta})$, where $\sum_j \mu_{j0}e^{X_j^T\beta}=e^{X_c\beta} \sum_j \mu_{j0}=\mu_{c0}e^{X_c\beta}$, assuming $Y_j$ are independent.   



### Likelihood and derivative 


Let $f(\beta)=\sum_c\{Y_c(X_c^T\beta)-\mu_{c0}e^{X_c^T\beta} \}$ be the objective function to be optimized over $\beta$. 
$\frac{\partial f(\beta)}{\partial \beta_i}=\sum_c (Y_c(X_{ic})-\mu_{c0}X_{ic}e^{X_c^T\beta})$, $i=1, \cdots, P$, and 

$\frac{\partial f(\beta)}{\partial \beta_0}=\sum_c (Y_c-\mu_{c0}e^{X_c^T\beta})$. 


## Predictive density 

Let $\tilde{y}$ be a new value, $Y$  be the observed data, the posterior predictive distribution of $\tilde{y}$ is 

\begin{equation}
p(\tilde{y})=\int p(\tilde{y}|\beta, Y)p(\theta|Y)d\beta. 
\end{equation}

For the Poisson regression model where $\beta$ is fixed, the log predictive density is, by (1) 
$l(\tilde{y}|\widehat{\beta})=\sum_jY_jlog(\widehat{\mu}_j)-\sum_j\widehat{\mu}_j$, where $\widehat{\mu}_j$ is the predicted mutation rate using the parameter estimated in the training set. Note $\widehat{\mu}_j=\mu_{j0}e^{X_j^T\widehat{\beta}}=0$ when $\mu_{j0}=0$, resulting in invalid predictive density for window $j$.  
